### PR TITLE
Move to Omnisharp lib 0.18.x

### PR DESCRIPTION
**Pull request by TylerLeonhardt**

### CODE REVIEW

File: src/PowerShellEditorServices/Server/PsesDebugServer.cs [hunk 1]
```diff
@@ -71,12 +73,9 @@ internal class PsesDebugServer : IDisposable
...
         public async Task StartAsync()
         {
-            _jsonRpcServer = await JsonRpcServer.From(options =>
+            _debugAdapterServer = await DebugAdapterServer.From(options =>
             {
-                options.Serializer = new DapProtocolSerializer();
-                options.Receiver = new DapReceiver();
-                options.LoggerFactory = _loggerFactory;
-                ILogger logger = options.LoggerFactory.CreateLogger("DebugOptionsStartup");
+                options.WithSerializer(new DapProtocolSerializer());
```
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Server/PsesDebugServer.cs:78]: this line won't be needed for 0.18.2

File: test/PowerShellEditorServices.Test.E2E/DebugAdapterProtocolMessageTests.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            Assert.True(PsesDebugAdapterClient.ServerSettings.SupportsFunctionBreakpoints);
+            Assert.True(PsesDebugAdapterClient.ServerSettings.SupportsHitConditionalBreakpoints);
+            Assert.True(PsesDebugAdapterClient.ServerSettings.SupportsLogPoints);
+            Assert.True(PsesDebugAdapterClient.ServerSettings.SupportsSetVariable);
+        }
+
+        [Fact]
+        public async Task CanLaunchScriptWithNoBreakpointsAsync()
+        {
+            string filePath = NewTestFile("'works' > \"$PSScriptRoot/testFile.txt\"");
```
Comment (david-driscoll) [inline test/PowerShellEditorServices.Test.E2E/DebugAdapterProtocolMessageTests.cs:55]: That's an integration test with assertions!

**Review by david-driscoll — APPROVED** [review]
Looks good to me, let me know if you have more questions!

**Review by rjmholt — COMMENTED** [review]
Oh, hmm, it didn't post my review when I wrote it... Ok now it's posting

File: src/PowerShellEditorServices/Services/DebugAdapter/Handlers/BreakpointHandlers.cs [hunk 3]
```diff
@@ -218,5 +117,72 @@ public async Task<SetBreakpointsResponse> Handle(SetBreakpointsArguments request
...
+
+            return new SetFunctionBreakpointsResponse
+            {
+                Breakpoints = updatedBreakpointDetails
+                    .Select(LspDebugUtils.CreateBreakpoint)
+                    .ToArray()
+            };
+        }
+
+        public Task<SetExceptionBreakpointsResponse> Handle(SetExceptionBreakpointsArguments request, CancellationToken cancellationToken)
```
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/DebugAdapter/Handlers/BreakpointHandlers.cs:161]: I've not changed the code. Just moved it to a different space. There isn't a `SupportsExceptionBreakpoints` capability so we have to handle this message... but it seems like it just silently does nothing... very odd.

File: src/PowerShellEditorServices/Services/DebugAdapter/Handlers/LaunchAndAttachHandler.cs [hunk 4]
```diff
@@ -207,45 +201,14 @@ public async Task<Unit> Handle(PsesLaunchRequestArguments request, CancellationT
             // debugging session
             _debugStateService.IsInteractiveDebugSession = string.IsNullOrEmpty(_debugStateService.ScriptToLaunch);
 
-            // Send the InitializedEvent so that the debugger will continue
+            // Sends the InitializedEvent so that the debugger will continue
             // sending configuration requests
-            _jsonRpcServer.SendNotification(EventNames.Initialized);
+            _debugStateService.ServerStarted.SetResult(true);
```
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/DebugAdapter/Handlers/LaunchAndAttachHandler.cs:206]: changed.

File: src/PowerShellEditorServices/Server/PsesDebugServer.cs [hunk 5]
```diff
@@ -97,45 +96,49 @@ public async Task StartAsync()
...
+                    .OnInitialized((server, request, response, cancellationToken) => {
+                        response.SupportsConditionalBreakpoints = true;
+                        response.SupportsConfigurationDoneRequest = true;
+                        response.SupportsFunctionBreakpoints = true;
+                        response.SupportsHitConditionalBreakpoints = true;
+                        response.SupportsLogPoints = true;
+                        response.SupportsSetVariable = true;
+
+                        return Task.CompletedTask;
+                    });
```
[the remark this answers is not in the document]
Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Server/PsesDebugServer.cs]: I added comments in a few places to explain what we're doing here... it's a bit confusing. @david-driscoll says it's cause we're weird 😛

**Review by rjmholt — APPROVED** [review]

File: src/PowerShellEditorServices/Server/PsesDebugServer.cs [hunk 6]
```diff
@@ -113,11 +113,15 @@ public async Task StartAsync()
...
                     .WithHandler<DebugEvaluateHandler>()
+                    // The OnInitialize delegate gets run when we first receive the _Initialize_ request:
+                    // https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Initialize
                     .OnInitialize(async (server, request, cancellationToken) => {
                         var breakpointService = server.GetService<BreakpointService>();
                         // Clear any existing breakpoints before proceeding
                         await breakpointService.RemoveAllBreakpointsAsync().ConfigureAwait(false);
                     })
+                    // The OnInitialized delegate gets run right before the server responds to the _Initialize_ request:
+                    // https://microsoft.github.io/debug-adapter-protocol/specification#Requests_Initialize
```
Comment (david-driscoll) [inline src/PowerShellEditorServices/Server/PsesDebugServer.cs:124]: 👍

File: src/PowerShellEditorServices/Services/DebugAdapter/Handlers/BreakpointHandlers.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (david-driscoll) [reply src/PowerShellEditorServices/Services/DebugAdapter/Handlers/BreakpointHandlers.cs:161]: https://microsoft.github.io/debug-adapter-protocol/specification#arrow_left-initialized-event
`frontend sends a ‘setExceptionBreakpoints’ request if one or more ‘exceptionBreakpointFilters’ have been defined (or if ‘supportsConfigurationDoneRequest’ is not defined or false)`

File: src/PowerShellEditorServices/Server/PsesDebugServer.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (david-driscoll) [reply src/PowerShellEditorServices/Server/PsesDebugServer.cs]: Naming things is hard... so I just stuck with the protocol naming for the events.

[the remark this answers is not in the document]
Comment (david-driscoll) [reply src/PowerShellEditorServices/Server/PsesDebugServer.cs]: I need to do some more documentation on how things work, it's on my list.

**Pull request merged**